phpBB/includes/functions_acp.php

```diff
@@ -367,83 +374,42 @@ function phpbb_build_cfg_template(array $tpl_type, string $key, &$new_ary, $conf
 		break;
 
 		case 'radio':
-			$tpl_type_cond = explode('_', $tpl_type[1]);
-			$type_no = $tpl_type_cond[0] != 'disabled' && $tpl_type_cond[0] != 'enabled';
-
-			$no_button = [
-				'type'		=> 'radio',
-				'name'		=> $name,
-				'value'		=> 0,
-				'checked'	=> !$new_ary[$config_key],
-				'label'		=> $type_no ? $language->lang('NO') : $language->lang('DISABLED'),
-			];
-
-			$yes_button = [
-				'id'		=> $key,
-				'type'		=> 'radio',
-				'name'		=> $name,
-				'value'		=> 1,
-				'checked'	=> (bool) $new_ary[$config_key],
-				'label'		=> $type_no ? $language->lang('YES') : $language->lang('ENABLED'),
-			];
-
-			$tpl = ['tag' => 'radio'];
-			if ($tpl_type_cond[0] == 'yes' || $tpl_type_cond[0] == 'enabled')
+			// Convert 'old' radio button parameters to json encoded string for BC
+			if (in_array($tpl_type[1], ['yes_no', 'enabled_disabled']))
 			{
-				$tpl['buttons'] = [$yes_button, $no_button];
+				$params = explode('_', $tpl_type[1]);
+				$tpl_type[1] = '{"' . $params[0] . '":1, "' . $params[1] . '":0}';
 			}
-			else
-			{
-				$tpl['buttons'] = [$no_button, $yes_button];
-			}
-		break;
-	}
 
-	return $tpl;
-}
-
-/**
-* Build configuration template for acp configuration pages
-*/
-function build_cfg_template($tpl_type, $key, &$new_ary, $config_key, $vars)
-{
-	global $module, $phpbb_dispatcher;
-
-	$tpl = '';
-	$name = 'config[' . $config_key . ']';
-
-	// Make sure there is no notice printed out for non-existent config options (we simply set them)
-	if (!isset($new_ary[$config_key]))
-	{
-		$new_ary[$config_key] = '';
-	}
+			$params = json_decode($tpl_type[1], true);
```

Comment:
Could you maybe change the radio options to a similar way on how it's done with generating the select dropdown? So that you can just pass an array of options for the radio buttons?

Reply:
The problem here is that the options are being set in a string within ACP module class. As for `select`, they're only generated using class method. But if this change is welcome I can do it like that (BC will be broken).